### PR TITLE
feat: adds js implementation of rabin chunker for windows and browser

### PR DESCRIPTION
**Pull request by achingbrain**
Ported from https://www.npmjs.com/package/rabin

### CODE REVIEW

**Comment by achingbrain** [comment]
Also fixes a bug with the native chunker that meant it was truncating files.

File: src/chunker/rabin.js [hunk 1]
```diff
@@ -30,30 +28,216 @@ module.exports = async function * rabinChunker (source, options) {
...
+
+    if (rabin.buffers.length) {
+      yield rabin.buffers.slice(0)
+    }
+  }
+}
+
+const jsRabin = () => {
+  // see https://github.com/datproject/rabin/blob/c0378395dc0a125ab21ac176ec504f9995b34e62/src/rabin.cc
+  class Rabin {
```
Comment (alanshaw) [inline src/chunker/rabin.js:80]: `npm i rabin.js`?

File: test/chunker-rabin.spec.js [hunk 2]
```diff
@@ -70,11 +70,10 @@ describe('chunker: rabin', function () {
...
+      ...defaultOptions,
       minChunkSize: KiB256 / 3,
       avgChunkSize: KiB256,
-      maxChunkSize: KiB256 + (KiB256 / 2),
-      window: 16,
-      polynomial: '0x3DF305DFB2A805'
+      maxChunkSize: KiB256 + (KiB256 / 2)
     }
 
     const chunks = await all(chunker([file], opts))
```
Comment (alanshaw) [inline test/chunker-rabin.spec.js:79]: Should we add a test to check that the output of JS rabin impl is the same as native rabin impl?

**Review by alanshaw — COMMENTED** [review]
This is really cool ❤️🚀

🙏 please next time can the @ipfs/wg-js-core have a chance to review before merge+release?

File: src/chunker/rabin.js (hunk 1 above)
Comment (achingbrain) [reply src/chunker/rabin.js:80]: Do you mean [rabinjs](https://www.npmjs.com/package/rabinjs)?  It has no repo on npm and only 5 downloads a week.

Not sure it's ready for production.

File: test/chunker-rabin.spec.js (hunk 2 above)
Comment (achingbrain) [reply test/chunker-rabin.spec.js:79]: The rabin tests run in node and the browser, one of them [tests the size of chunks emitted](https://github.com/ipfs/js-ipfs-unixfs-importer/blob/master/test/chunker-rabin.spec.js#L23-L50) so I thought that was sufficient.

I was adding interop tests for more comprehensive testing but then @hugomrdias compiled the C version to WASM in [rabin-wasm](https://github.com/hugomrdias/rabin-wasm) so may wait for that to be ready.

Comment (achingbrain) [reply test/chunker-rabin.spec.js:79]: Interop tests are at ipfs/interop#74 - though as ever the build is failing due to unrelated tests that depend on networking.

**Pull request merged**